store/store.go

```diff
@@ -1228,6 +1240,40 @@ var applyDelta = func(name string, deltaPath string, deltaInfo *snap.DeltaInfo)
 	return targetSnapPath, nil
 }
 
+// downloadAndApplyDelta downloads and then applies the delta to the current snap.
+func (s *Store) downloadAndApplyDelta(name, targetFn string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (err error) {
+	deltaInfo := &downloadInfo.Deltas[0]
+	workingDir := filepath.Dir(targetFn)
+
+	deltaPath, err := s.downloadDelta(name, workingDir, downloadInfo, pbar, user)
```

Comment:
Thanks @niemeyer . I believe I've implemented as above (and switched s/targetFn/targetPath/ as suggested below by @mvo5